.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,33 @@
+name: 'CodeQL'
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  schedule:
+    - cron: '35 21 * * 5'
```

Comment:
question: why this cron value (each friday at 21h35) ?

Reply:
an why on each pull_request is not enough ?

Reply:
CodeQL is quite slow, and relies on CVEs.
You can spin it more often, but I'm not sure it's really more comfortable.

Ditto for running it on all the PRs, this is possible

Reply:
🙏 Thank you for all @flemzord . I did not know CodeQL. Very interesting. I merge. 

PS : instead of CVEs, you meant CWEs ([Common Weakness Enumeration](https://cwe.mitre.org/)), right ?